src/app/detailed-viewer/detailed-viewer.component.ts

```diff
@@ -0,0 +1,72 @@
+import { AfterViewInit, Component, OnDestroy, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Subscription } from 'rxjs';
+import { GithubUser } from '../core/models/github-user.model';
+import { Phase } from '../core/models/phase.model';
+import { GithubService } from '../core/services/github.service';
+import { LoggingService } from '../core/services/logging.service';
+import { MilestoneService } from '../core/services/milestone.service';
+import { PhaseService } from '../core/services/phase.service';
+
+/**
+ * Responsible for structuring the page as well as passing data between components.
+ * Also responsible for retrieving Issues and PRs from the repos and classifying them
+ * into "Created" and "Assigned" PRs and Issues to be displayed by the 4 ProfileListComponents
+ */
+@Component({
+  selector: 'app-detailed-viewer',
+  templateUrl: './detailed-viewer.component.html',
+  styleUrls: ['./detailed-viewer.component.css']
+})
+export class DetailedViewerComponent implements OnInit, OnDestroy, AfterViewInit {
+  public user: GithubUser = undefined;
+
+  private userSubscription: Subscription;
+
+  /** Observes for any change in repo*/
+  repoChangeSubscription: Subscription;
+  issueChangeSubscription: Subscription;
+
+  constructor(
+    private phaseService: PhaseService,
+    public githubService: GithubService,
+    public milestoneService: MilestoneService,
+    private route: ActivatedRoute,
+    private router: Router,
+    private logger: LoggingService
+  ) {
+    this.repoChangeSubscription = this.phaseService.repoChanged$.subscribe((newRepo) => {
+      this.initialize();
+    });
+  }
+
+  ngOnInit() {
+    if (this.route.snapshot.paramMap.get('name') === undefined) {
```

Comment:
The return result could be null, which is different from undefined.

Perhaps we can use the falsy value

Reply:
Yes, thanks for pointing out, I misread the return type of paremMap.get()
Luckily a separate check was done before this so this is actually never reached, it's more like defensive programming here.